Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -1,95 +1,183 @@
         public static int[] InsertFirst(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            int[] newArray = new int[array.Length + 1];
+
+            if (IsNullOrEmpty(array))
+            {
+                newArray[0] = number;
+                return newArray;
+            }
+
+            newArray[0] = number;
+
+            for (int i = 1; i < newArray.Length; i++)
+            {
+                newArray[i] = array[i - 1];
+            }
+
+            return newArray;
         }
 
-        /// <summary>
-        /// Inserts a new array element at the end.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="number">Number to be added.</param>
-        /// <returns>A new array with element added in the end of array. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertLast(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            int[] newArray = new int[array.Length + 1];
+
+            if (IsNullOrEmpty(array))
+            {
+                newArray[0] = number;
+                return newArray;
+            }
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                newArray[i] = array[i];
+            }
+
+            newArray[newArray.Length - 1] = number;
+
+            return newArray;
         }
 
-        /// <summary>
-        /// Inserts a new array element at the specified index.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="number">Number to be added.</param>
-        /// <param name="index">Index at which array element should be added.</param>
-        /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            int[] newArray = new int[array.Length + 1];
+
+            if (IsNullOrEmpty(array))
+            {
+                newArray[0] = number;
+                return newArray;
+            }
+
+            for (int i = 0; i < index; i++)
+            {
+                newArray[i] = array[i];
+            }
+
+            newArray[index] = number;
+
+            for (int i = index + 1; i < newArray.Length; i++)
+            {
+                newArray[i] = array[i - 1];
+            }
+
+            return newArray;
+        }
+
+        internal static bool Sort()
+        {
+            throw new NotImplementedException();
+        }
+
+        private static bool IsNullOrEmpty(int[] array)
+        {
+            if (array.Length == 0 || array == null)
```

Comment:
Flip this around. `array.Length` will throw a NullReferenceException. Remember that `||` is short-circuiting.